src/java.base/share/classes/java/lang/Runtime.java

```diff
@@ -276,9 +278,11 @@ public boolean removeShutdownHook(Thread hook) {
      * to finish if it is already in progress. This method never returns normally.
      *
      * @apiNote
-     * This method should be used with extreme caution. Using it may circumvent or disrupt
-     * any cleanup actions intended to be performed by shutdown hooks, possibly leading to
-     * data corruption.
+     * This method should be used with extreme caution &mdash; halting all threads
+     * can have many unexpected results. For instance, it may circumvent cleanup
+     * actions intended to be performed by shutdown hooks,
+     * {@linkplain Thread.UncaughtExceptionHandler uncaught exception handlers},
+     * finally blocks, or try-with-resources blocks. This can lead to data corruption.
```

Comment:
Since the examples of the consequences of halting are covered above in "JVM Termination" and are getting a bit involved, I'd suggest not trying to list the concrete consequences here, and referring the reader (again) to the "JVM Termination" section.